### DISCUSSION TITLE
iOS SDKs (for *buildapp.yml*)

### DISCUSSION THREAD

**Discussion by arichornlover** [arichornlover/uYouEnhanced#740] [General]
Hello! I’m here to help y’all find which SDKs that can be used when building **uYouEnhanced**!
look below to see which SDK is available when modifying the “iOS SDK Version” number value in Build and Release uYouEnhanced actions.

# iOS SDKs List

| SDK Name & Version | Compiling IPA Compatibility | YouTube Versions Compatibility |
|----------------------|-----------------------------|------------------------|
| iPhoneOS **14.0** | works successfully ✅ | supports up to **v19.20.2** |
| iPhoneOS **14.1** | works successfully ✅ | supports up to **v19.20.2** |
| iPhoneOS **14.2** | works successfully ✅ | supports up to **v19.20.2** |
| iPhoneOS **14.3** | works successfully ✅ | supports up to **v19.20.2** |
| iPhoneOS **14.4** | works successfully ✅ | supports up to **v19.20.2** |
| iPhoneOS **14.5** | works successfully ✅ | supports up to **v19.20.2** |
| iPhoneOS **15.2** | works successfully ✅ | supports up to **v20.21.6** |
| iPhoneOS **15.5** | works successfully ✅ | supports up to **v20.21.6** |
| iPhoneOS **16.1** | works successfully ✅ | supports up to **v21.33.6** |
| iPhoneOS **16.2** | works successfully ✅ | supports up to **v21.33.6** |
| iPhoneOS **16.4** | works successfully ✅ | supports up to **v21.33.6** |
| iPhoneOS **16.5** - used by uYouPlus | works successfully ✅ | supports up to **v21.33.6** |
| iPhoneOS **17.0** | works successfully ✅ | supports up to the **latest** |
| iPhoneOS **17.0.2** | works successfully ✅ | supports up to the **latest** |
| iPhoneOS **17.0.3** | works successfully ✅ | supports up to the **latest** |
| iPhoneOS **17.1** | works successfully ✅ | supports up to the **latest** |
| iPhoneOS **17.2** | works successfully ✅ | supports up to the **latest** |
| iPhoneOS **17.3.1** | works successfully ✅ | supports up to the **latest** |
| iPhoneOS **17.4** | works successfully ✅ | supports up to the **latest** |
| iPhoneOS **17.5** | works successfully ✅ | supports up to the **latest** |
| iPhoneOS **18.0** | works successfully ✅ | supports up to the **latest** |
| iPhoneOS **18.1** | works successfully ✅ | supports up to the **latest** |
| iPhoneOS **18.2** | works successfully ✅ | supports up to the **latest** |
| iPhoneOS **18.3** | works successfully ✅ | supports up to the **latest** |
| iPhoneOS **18.4** | works successfully ✅ | supports up to the **latest** |
| iPhoneOS **18.5** | works successfully ✅ | supports up to the **latest** |
| iPhoneOS **18.6** - used by uYouEnhanced | works successfully ✅ | supports up to the **latest** |
| iPhoneOS **26.0** | works successfully ✅ | supports up to the **latest** |
| iPhoneOS **26.1** | works successfully ✅ | supports up to the **latest** |
| iPhoneOS **26.2** | works successfully ✅ | supports up to the **latest** |
| iPhoneOS **26.3** | works successfully ✅ | supports up to the **latest** |
| iPhoneOS **26.4** | works successfully ✅ | supports up to the **latest** |
| iPhoneOS **26.4.1** | works successfully ✅ | supports up to the **latest** |
| iPhoneOS **26.5** | works successfully ✅ | supports up to the **latest** |
| iPhoneOS **27.0** | doesn’t work successfully 🚫 | supports up to the **latest** |

**Comment by mucookul** [comment]
The `SDK not found` is caused because the Makefile specifies 16.4, and for some reason when using 16.5, it won't auto upgrade, changing the makefile to 16.5 does result in a successful build I think.

Edit: 16.5 works just fine on the latest youtube ipa.

**Reply by arichornlover** [reply]
> The `SDK not found` is caused because the Makefile specifies 16.4, and for some reason when using 16.5, it won't auto upgrade, changing the makefile to 16.5 does result in a successful build I think.
> 
> Edit: 16.5 works just fine on the latest youtube ipa.

it's also labeled as that because it doesn't exist in the **arichornlover/sdks** repo

**Reply by mucookul** [reply]
Why don't you add it though, I would assume it's the most popular sdk with the most support as it's the default for a normal theos installation.

**Reply by mucookul** [reply]
Also that sdk-extractor is really nice

**Reply by arichornlover** [reply]
> Why don't you add it though, I would assume it's the most popular sdk with the most support as it's the default for a normal theos installation.

I would but I gotta somehow move it over to the other repo, if I had the .zip of the SDK.

**Reply by arichornlover** [reply]
> Also that sdk-extractor is really nice

Thanks! It took time to make but it’s the best thing to use when it comes to extracting sdks to the repository! 😄

**Comment by aricloverEXTRA** [comment]
All iOS 17 SDKS will now work on uYouEnhanced!

**Comment by ehfleet** [comment]
Is this still the case for Youtube 21.07.4? to use iOS SDK 17.5?

**Reply by aricloverEXTRA** [reply]
Of course any SDK that ranges from iOS 16, iOS 17 & iOS 18.
